src/request/components/request-detail-panel-execution.jsx

```diff
@@ -70,6 +74,128 @@ var RequestHeaders = React.createClass({
+            <div>Mongo DB URI: {mongoOperation.payload.connectionHost}:{mongoOperation.payload.connectionPort}/{mongoOperation.payload.database}/{mongoOperation.payload.collection} </div>
+            <div>options: {JSON.stringify(mongoOperation.payload.options)}</div>
+        </div>;
+        return details;
+    }
+});
+ 
+var MongoReadCommand = React.createClass({
+  mixins: [MongoCommandMixin],
+  
+  getTitle: function() {
+      return 'MongoDB - Read documents';
+  },
+  
+  getDetails: function() {
+      var mongoOperation = this.props.mongoOperation;
+      var details = 
+          <div>
+              <div>Operation: {mongoOperation.payload.operation}</div>
+              <div>Mongo DB URI: {mongoOperation.payload.connectionHost}:{mongoOperation.payload.connectionPort}/{mongoOperation.payload.database}/{mongoOperation.payload.collection} </div>
+              <div>Query: {JSON.stringify(mongoOperation.payload.query)}</div>
+              <div>options: {JSON.stringify(mongoOperation.payload.options)}</div>
+          </div>;
+      return details;
+  }
+});
+
+var MongoUpdateCommand = React.createClass({
+    mixins: [MongoCommandMixin],
+    
+    getTitle: function() {
+        return 'MongoDB - Updated ' + this.props.mongoOperation.payload.modifiedCount +  ' documents';
+    },
+    
+    getDetails: function() {
+        var mongoOperation = this.props.mongoOperation;
+        var details =
+        <div> 
+            <div>Operation: {mongoOperation.payload.operation}</div>
+            <div>Mongo DB URI: {mongoOperation.payload.connectionHost}:{mongoOperation.payload.connectionPort}/{mongoOperation.payload.database}/{mongoOperation.payload.collection} </div>
+            <div>matched count: {mongoOperation.payload.matchedCount}</div>
+            <div>modified count: {mongoOperation.payload.modifiedCount}</div>
+            <div>upserted count: {mongoOperation.payload.upsertedCount}</div>
+            <div>query: {JSON.stringify(mongoOperation.payload.query)}</div>
+            <div>options: {JSON.stringify(mongoOperation.payload.options)}</div>
+        </div>;
+        return details;
+    }
+});
+ 
+var MongoDeleteCommand = React.createClass({
+    mixins: [MongoCommandMixin],
+    
+    getTitle: function() {
+      return 'MongoDB - Deleted ' + this.props.mongoOperation.payload.count +  ' documents';
+    },
+    
+    getDetails: function() {
+        var mongoOperation = this.props.mongoOperation;
+        var details =
+        <div> 
+            <div>Operation: {mongoOperation.payload.operation}</div>
+            <div>Mongo DB URI: {mongoOperation.payload.connectionHost}:{mongoOperation.payload.connectionPort}/{mongoOperation.payload.database}/{mongoOperation.payload.collection} </div>
+            <div>query: {JSON.stringify(mongoOperation.payload.query)}</div>
+            <div>options: {JSON.stringify(mongoOperation.payload.options)}</div>
+        </div>;
```

Comment:
If the way we are displaying this data is as a simple list, when we switch over to "data command" strategy (where we have templates bound to message types), we could have a view model defined which has the `title` and the `value`. That way we could just iterate through. For more complex cases, we could have an actual component. Again this can be done in the future.